pkg/skaffold/watch/triggers.go

```diff
@@ -40,20 +40,25 @@ type Trigger interface {
 }
 
 // NewTrigger creates a new trigger.
-func NewTrigger(opts *config.SkaffoldOptions) (Trigger, error) {
-	switch strings.ToLower(opts.Trigger) {
+func NewTrigger(runctx *runcontext.RunContext) (Trigger, error) {
+	switch strings.ToLower(runctx.Opts.Trigger) {
 	case "polling":
 		return &pollTrigger{
-			Interval: time.Duration(opts.WatchPollInterval) * time.Millisecond,
+			Interval: time.Duration(runctx.Opts.WatchPollInterval) * time.Millisecond,
 		}, nil
 	case "notify":
 		return &fsNotifyTrigger{
-			Interval: time.Duration(opts.WatchPollInterval) * time.Millisecond,
+			Interval: time.Duration(runctx.Opts.WatchPollInterval) * time.Millisecond,
 		}, nil
 	case "manual":
 		return &manualTrigger{}, nil
+	case "api":
+		return &apiTrigger{
+			Trigger:  runctx.Trigger,
+			Interval: time.Duration(runctx.Opts.WatchPollInterval) * time.Millisecond,
```

Comment:
shd we remove this from apiTrigger since Interval is not used anymore?

Reply:
ah yeah good call, done